src/Test/Utilities/Portable/TargetFrameworkUtil.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using static TestReferences;
+
+namespace Roslyn.Test.Utilities
+{
+    public enum TargetFramework
+    {
+        /// <summary>
+        /// Explicit pick a target framework that has no references
+        /// </summary>
+        None,
+
+        NetStandard20,
+        WinRT,
+
+        /// <summary>
+        /// Eventually this will be deleted and replaced with NetStandard20. Short term this creates the "standard"
+        /// API set across destkop and coreclr 
+        /// </summary>
+        Standard,
+
+        /// <summary>
+        /// This is represents the set of tests which must be mscorlib40 on desktop but full net standard on coreclr.
+        /// </summary>
+        StandardCompat,
+
+        // The flavors of mscorlib we support + extending them with LINQ and dynamic.
+        Mscorlib40,
+        Mscorlib40Extended, 
+        Mscorlib45,
+        Mscorlib45Extended,
+        Mscorlib46,
+        Mscorlib46Extended,
+    }
+
+    public static class TargetFrameworkUtil
+    {
+        public static readonly ImmutableArray<MetadataReference> NoneReferences = ImmutableArray<MetadataReference>.Empty;
+        public static readonly ImmutableArray<MetadataReference> Mscorlib40References = ImmutableArray.Create(TestBase.MscorlibRef);
+        public static readonly ImmutableArray<MetadataReference> Mscorlib40ExtendedReferences = ImmutableArray.Create(TestBase.MscorlibRef, TestBase.SystemRef, TestBase.SystemCoreRef, TestBase.ValueTupleRef, TestBase.SystemRuntimeFacadeRef);
+        public static readonly ImmutableArray<MetadataReference> Mscorlib45References = ImmutableArray.Create(TestBase.MscorlibRef_v4_0_30316_17626);
+        public static readonly ImmutableArray<MetadataReference> Mscorlib45ExtendedReferences = ImmutableArray.Create(TestBase.MscorlibRef_v4_0_30316_17626, TestBase.SystemRef, TestBase.SystemCoreRef, TestBase.ValueTupleRef, TestBase.SystemRuntimeFacadeRef);
+        public static readonly ImmutableArray<MetadataReference> Mscorlib46References = ImmutableArray.Create(TestBase.MscorlibRef_v46);
+        public static readonly ImmutableArray<MetadataReference> Mscorlib46ExtendedReferences = ImmutableArray.Create(TestBase.MscorlibRef_v46, TestBase.SystemRef_v46, TestBase.SystemCoreRef_v46, TestBase.ValueTupleRef, TestBase.SystemRuntimeFacadeRef);
+        public static readonly ImmutableArray<MetadataReference> NetStandard20References = ImmutableArray.Create<MetadataReference>(NetStandard20.NetStandard, NetStandard20.MscorlibRef, NetStandard20.SystemRuntimeRef, NetStandard20.SystemDynamicRuntimeRef);
+        public static readonly ImmutableArray<MetadataReference> WinRTReferences = ImmutableArray.Create(TestBase.WinRtRefs);
+        public static readonly ImmutableArray<MetadataReference> StandardReferences = CoreClrShim.IsRunningOnCoreClr ? NetStandard20References : Mscorlib46ExtendedReferences;
```

Comment:
I thought we were moving to a single reference set everywhere, only swapping out the runtime assemblies?

Reply:
That is the eventual plan but hard to do in this PR because of PEVerify. Until we move to ILVerify I need to keep desktop tests compiling against mscorlib so that our PEverify code can function.